woocommerce/woocommerce-ios
# [Order Fulfillment][Part 4] Add Tracking Section to Review Order screen

Part 4 of #4136. 
For more context, please take a look at [Part 1](https://github.com/woocommerce/woocommerce-ios/pull/4534), [Part 2](https://github.com/woocommerce/woocommerce-ios/pull/4535), [Part 3](https://github.com/woocommerce/woocommerce-ios/pull/4539), [Part 5](https://github.com/woocommerce/woocommerce-ios/pull/4567).

# Description
This PR follows up #4539 to add another section for Shipment Tracking on Review Order screen. There are nearly 600 lines of code changes and I blame my generated `Copiable` and unit tests for that.

# Solution
- The view model was updated to config tracking section for the review order table view with conditions:
  - There should be no non-refunded shipping labels, otherwise the whole Tracking section is hidden.
  - Shipment & Tracking section should be installed to the store, otherwise Add Tracking row is hidden.
  - There should be at least one shipment tracking added for the order, otherwise no tracking row is shown.
- The view controller was updated to config UI styling of the tracking and add tracking row, as well as handle the actions on these rows. Tracking rows allow 3 actions:
  - View tracking details on a web view
  - Copy tracking number
  - Delete tracking.
- The view controller and view model were updated to reload the table view when there's change in either add-ons, shipping labels or shipment tracking.

# Demo
<img src="https://user-images.githubusercontent.com/5533851/124591044-083afe80-de86-11eb-923f-f2123e5e8d17.gif" width=500 />

# Testing
- Navigate to Orders tab
- Select an order with In Progress status
- Select Mark Order Complete
- You should then be navigated to Review Order screen
- You should find Tracking section if conditions in Solution are met.
- When Add Tracking button is tapped, Add Tracking screen is presented. When new tracking is added and Add Tracking screen is dismissed, Review Order screen should show the new tracking immediately.
- Copying tracking code should work properly and include haptic feedback
- Viewing tracking detail should navigate to a web view of the specified tracking company with the tracking code.
- Delete tracking should remove the tracking properly.

Update release notes:
